net/haproxy/src/opnsense/service/templates/OPNsense/HAProxy/haproxy.conf

```diff
@@ -999,6 +1036,7 @@ global
 {% if helpers.exists('OPNsense.HAProxy.general.tuning.maxDHSize') %}
     tune.ssl.default-dh-param   {{OPNsense.HAProxy.general.tuning.maxDHSize}}
 {% endif %}
+    tune.ssl.capture-buffer-size 512
```

Comment:
Could you please make this option configurable?

I'd suggest to add two new options for this:
`Enable Capture Buffer` (default: disabled)
`Capture Buffer Size` (default: 512)

These should be added to `Services: HAProxy: Settings -> Global Parameters`.

Reply:
OK, but SNI parsing depends on the capture buffer being enabled to work, or it fails silently.
I guess this dependency should be added to the help text in both places then.
Maybe a warning during template generation could also be added.